8.17.6 release notes (#6801)
elastic/security-docs: docs/release-notes/8.17.asciidoc

[[release-notes-header-8.17.0]]
== 8.17

[discrete]
[[release-notes-8.17.6]]
=== 8.17.6

[discrete]
[[enhancements-8.17.6]]
==== Enhancements
* Allows {elastic-defend} users to opt out of event-driven Memory Protection scanning using the advanced policy ({kibana-pull}218354[#218354]).

> * Makes minor improvements to {elastic-defend}'s' CPU usage for systems with very high event volumes.

Reviewer: Sorry bad label on this one.  It didn't land in 8.17.6.
```suggestion
```

[discrete]
[[bug-fixes-8.17.6]]
==== Fixes
* Fixes a bug that caused installed prebuilt detection rules to upgrade to their latest available versions when you installed a new {elastic-defend} integration or {agent} policy ({kibana-pull}217959[#217959]).

> * Avoids an `IRQL_NOT_LESS_EQUAL` https://learn.microsoft.com/en-us/windows-hardware/drivers/debugger/bug-checks--blue-screens-[bugcheck] in the {elastic-defend} driver due to an interaction with Trellix Access Protection (`mfehidk.sys`). This issue can occur when `elastic-endpoint-driver.sys` calls https://learn.microsoft.com/en-us/windows/win32/api/fwpmu/nf-fwpmu-fwpmtransactionbegin0[`FwpmTransactionBegin0`] to initialize its network driver. `FwpmTransactionBegin0` performs a synchronous RPC call to the user-mode Base Filtering Engine service. Trellix's driver intercepts this service's operations, causing `FwpmTransactionBegin0` to hang. This hang prevents Defend's driver from properly initializing in a timely manner. Subsequent system activity can invoke {elastic-defend}'s driver before it has fully initialized, leading to a `IRQL_NOT_LESS_EQUAL` bugcheck.

Reviewer: ```suggestion
* Avoids an `IRQL_NOT_LESS_EQUAL` https://learn.microsoft.com/en-us/windows-hardware/drivers/debugger/bug-checks--blue-screens-[bugcheck] in the {elastic-defend} driver due to an interaction with Trellix Access Protection (`mfehidk.sys`). This issue can occur when `elastic-endpoint-driver.sys` calls https://learn.microsoft.com/en-us/windows/win32/api/fwpmu/nf-fwpmu-fwpmtransactionbegin0[`FwpmTransactionBegin0`] to initialize its network driver. `FwpmTransactionBegin0` performs a synchronous RPC call to the user-mode Base Filtering Engine service. Trellix's driver intercepts this service's operations, causing `FwpmTransactionBegin0` to hang or slow significantly. This delay prevents Defend's driver from properly initializing in a timely manner. Subsequent system activity can invoke {elastic-defend}'s driver before it has fully initialized, leading to a `IRQL_NOT_LESS_EQUAL` bugcheck.  Affected users who are unable to upgrade can prevent this issue from occurring by either disabling Trellix Access Protection or adding a https://docs.trellix.com/bundle/endpoint-security-10.6.0-threat-prevention-client-interface-reference-guide-windows/page/GUID-6AC245A1-5E5D-4BAF-93B0-FE7FD33571E6.html[Trellix Access Protection exclusion] for the Base Filtering Engine service (C:\Windows\System32\svchost.exe).  This issue affects Defend versions 8.16.0-8.16.6, 8.17.0-8.17.5, 8.18.0, and 9.0.0.
```

Author: Minor adjustments to spacing, text, and grammar.

```suggestion
* Avoids an `IRQL_NOT_LESS_EQUAL` https://learn.microsoft.com/en-us/windows-hardware/drivers/debugger/bug-checks--blue-screens-[bugcheck] in the {elastic-defend} driver due to an interaction with Trellix Access Protection (`mfehidk.sys`). This issue can occur when `elastic-endpoint-driver.sys` calls https://learn.microsoft.com/en-us/windows/win32/api/fwpmu/nf-fwpmu-fwpmtransactionbegin0[`FwpmTransactionBegin0`] to initialize its network driver. `FwpmTransactionBegin0` performs a synchronous RPC call to the user-mode Base Filtering Engine service. Trellix's driver intercepts this service's operations, causing `FwpmTransactionBegin0` to hang or slow significantly. This delay prevents Defend's driver from properly initializing in a timely manner. Subsequent system activity can invoke {elastic-defend}'s driver before it has fully initialized, leading to a `IRQL_NOT_LESS_EQUAL` bugcheck. 
+
If you're unable to upgrade, you can prevent this issue from occurring by either disabling Trellix Access Protection or adding a https://docs.trellix.com/bundle/endpoint-security-10.6.0-threat-prevention-client-interface-reference-guide-windows/page/GUID-6AC245A1-5E5D-4BAF-93B0-FE7FD33571E6.html[Trellix Access Protection exclusion] for the Base Filtering Engine service (C:\Windows\System32\svchost.exe). This issue affects {elastic-defend} versions 8.16.0-8.16.6, 8.17.0-8.17.5, 8.18.0, and 9.0.0.
```

> * Ensures that {elastic-defend} prevents unbounded kernel non-paged memory growth during extremely high event load situations on Windows.

Reviewer: Technically correct as-is, but we could add more information to help users.

```suggestion
* Resolves an unbounded kernel non-paged memory growth issue in {elastic-defend}'s kernel driver during extremely high event load situations on Windows.  Systems affected by this issue would slow down or become unreponsive until the triggering event load (e.g. network activity) subsided.  We are only aware of this issue occurring on very busy Windows Server systems running Defend 8.16.0 and later.
```

> * Allows {elastic-defend} to detect and recover from a corrupt persistent cache database.  Previously, such databases would be unusable, effectively disabling the 

Reviewer: ```suggestion
* Allows {elastic-defend} to detect and recover from a corrupt persistent cache database.  Previously, such databases would be unusable, effectively disabling the persistent cache.
```

[[release-notes-header-8.17.0]]
== 8.17

[discrete]
[[release-notes-8.17.6]]
=== 8.17.6

[discrete]
[[enhancements-8.17.6]]
==== Enhancements
* Allows {elastic-defend} users to opt out of event-driven Memory Protection scanning using the advanced policy ({kibana-pull}218354[#218354]).

[discrete]
[[bug-fixes-8.17.6]]
==== Fixes

> * Fixes a bug that caused installed prebuilt detection rules to upgrade to their latest available versions when you installed a new {elastic-defend} integration or {agent} policy ({kibana-pull}217959[#217959]).

Reviewer 2: ```suggestion
* Fixes a bug that caused installed prebuilt detection rules to upgrade to their latest versions when you installed a new {elastic-defend} integration or {agent} policy ({kibana-pull}217959[#217959]).
```

Author: I do want to leave "available" here because it's an important qualifier in the sentence. My understanding is that there could be situations where the latest versions of a rule aren't available because users haven't updated their rules in a while.

> * Avoids an `IRQL_NOT_LESS_EQUAL` https://learn.microsoft.com/en-us/windows-hardware/drivers/debugger/bug-checks--blue-screens-[bugcheck] in the {elastic-defend} driver due to an interaction with Trellix Access Protection (`mfehidk.sys`). This issue can occur when `elastic-endpoint-driver.sys` calls https://learn.microsoft.com/en-us/windows/win32/api/fwpmu/nf-fwpmu-fwpmtransactionbegin0[`FwpmTransactionBegin0`] to initialize its network driver. `FwpmTransactionBegin0` performs a synchronous RPC call to the user-mode Base Filtering Engine service. Trellix's driver intercepts this service's operations, causing `FwpmTransactionBegin0` to hang or slow significantly. This delay prevents {elastic-defend}'s driver from properly initializing in a timely manner. Subsequent system activity can invoke {elastic-defend}'s driver before it has fully initialized, leading to a `IRQL_NOT_LESS_EQUAL` bugcheck. 

Reviewer 2: The Microsoft URL leads to a 404 error. Are we sure it's the right one?

Author: Yeah, this is the correct link. For some reason, the double dashes at the end of the URL is being changed from `bug-checks--blue-screens-` to `bug-checks—%E2%80%8Bblue-screens-` when you click on the linked text in the HTML output.

Author: Attempting to escape the double-dashes.

```suggestion
* Avoids an `IRQL_NOT_LESS_EQUAL` https://learn.microsoft.com/en-us/windows-hardware/drivers/debugger/bug-checks\--blue-screens-[bugcheck] in the {elastic-defend} driver due to an interaction with Trellix Access Protection (`mfehidk.sys`). This issue can occur when `elastic-endpoint-driver.sys` calls https://learn.microsoft.com/en-us/windows/win32/api/fwpmu/nf-fwpmu-fwpmtransactionbegin0[`FwpmTransactionBegin0`] to initialize its network driver. `FwpmTransactionBegin0` performs a synchronous RPC call to the user-mode Base Filtering Engine service. Trellix's driver intercepts this service's operations, causing `FwpmTransactionBegin0` to hang or slow significantly. This delay prevents {elastic-defend}'s driver from properly initializing in a timely manner. Subsequent system activity can invoke {elastic-defend}'s driver before it has fully initialized, leading to a `IRQL_NOT_LESS_EQUAL` bugcheck. 
```

+

> If you're unable to upgrade, you can prevent this issue from occurring by either disabling Trellix Access Protection or adding a https://docs.trellix.com/bundle/endpoint-security-10.6.0-threat-prevention-client-interface-reference-guide-windows/page/GUID-6AC245A1-5E5D-4BAF-93B0-FE7FD33571E6.html[Trellix Access Protection exclusion] for the Base Filtering Engine service (`C:\Windows\System32\svchost.exe`). This issue affects {elastic-defend} versions 8.16.0-8.16.6, 8.17.0-8.17.5, 8.18.0, and 9.0.0.

Reviewer 2: ```suggestion
If you can't upgrade, you can prevent this issue from occurring by either disabling Trellix Access Protection or adding a https://docs.trellix.com/bundle/endpoint-security-10.6.0-threat-prevention-client-interface-reference-guide-windows/page/GUID-6AC245A1-5E5D-4BAF-93B0-FE7FD33571E6.html[Trellix Access Protection exclusion] for the Base Filtering Engine service (`C:\Windows\System32\svchost.exe`). This issue affects {elastic-defend} versions 8.16.0-8.16.6, 8.17.0-8.17.5, 8.18.0, and 9.0.0.
```

> * Resolves an unbounded kernel non-paged memory growth issue in {elastic-defend}'s kernel driver during extremely high event load situations on Windows. Systems affected by this issue would slow down or become unreponsive until the triggering event load (for example, network activity) subsided. We are only aware of this issue occurring on very busy Windows Server systems running {elastic-defend} 8.16.0 and later.

Reviewer 2: ```suggestion
* Resolves an unbounded kernel non-paged memory growth issue in {elastic-defend}'s kernel driver during extremely high event load situations on Windows. Systems affected by this issue would slow down or become unresponsive until the triggering event load (for example, network activity) subsided. We are only aware of this issue occurring on very busy Windows Server systems running {elastic-defend} 8.16.0 and later.
```

> * Allows {elastic-defend} to detect and recover from a corrupt persistent cache database.  Previously, such databases would be unusable, effectively disabling the persistent cache.

Reviewer 2: ```suggestion
* Allows {elastic-defend} to detect and recover from a corrupt persistent cache database.  Previously, such databases would be unusable, effectively turning off the persistent cache.
```
